### PR TITLE
regularize and validate of paths in the /mount API

### LINKED ISSUE #739: Mount API sometimes requires a trailing slash in the path, sometimes does not
- `DELETE` requires the slash - `There is no mount point at /data2`
- `GET` does not - `There is no mount point at /data2/`

I'm not sure on `POST` and `PUT` yet, will update the issue when I get to those.

### PR DESCRIPTION
**Pull request by mossprescott**
Fixes #739.

Enforce a consistent interpretation of paths for mounts. MongoDB mounts should
always use a directory path, and in the API those paths always have a trailing
slash.

Test coverage for all requests to /mount, and several small fixes:
- make DELETE idempotent
- use X-File-Name in POST
- use application/json Content-Type
- fix response text in a couple of cases

In the admin UI, always add the trailing slash.

### CODE REVIEW

**Comment by mossprescott** [comment]
Note: this is based on the commit in #770, so should be merged after that.

**Comment by jdegoes** [comment]
After Googling, the convention `X-Filename` has 6 million hits, while the convention `X-File-Name` has 500k, so I'd be in favor of switching to the former.

Review / merge by @sellout

**Comment by mossprescott** [comment]
Happy to make the switch, although my count was more like 13k to 28k the other way. What are you typing to google?

**Comment by jdegoes** [comment]
I could have gotten it wrong:

```
+"X-File-Name"

+"X-Filename"
```

**Comment by mossprescott** [comment]
Weird. I was just doing "X-Filename", and adding the leading + gets many more results, including some that just have "filename", or even "X-File-Name". Google is truly a wonderful and mysterious thing.

In any case, I don't feel strongly about one spelling or the other.

**Comment by jdegoes** [comment]
Damn Google. There was a day when it actually paid attention to what you wanted to search for. Now it knows better than all of us and just gives us whatever the hell it feels like.

**Comment by mossprescott** [comment]
Now with a cleaner diff since the other PR was merged.

**Comment by sellout** [comment]
Can we accept _both_ headers (but document/prefer one)?

Also, I could see us accepting POST with no header and content that looks like
```json
{ "baz": { "mongodb": … },
  "zap": { "mongodb": … } }
```
which would add subdirs “baz” and “zap” to the path in the request. But I don’t know how RFC-compliant that is.

Upside: append multiple endpoints in a single request, no `X-` header
Downside: dealing with partial failure, … probably others

**Comment by jdegoes** [comment]
Well, I'm about to add a ticket for a POST API which adds the JSON into the collection. So you will POST on a data file.

I'm not sure that POSTing on a directory makes sense until if / when we have a guarantee that directories actually exist...

File: README.md [hunk 1]
```diff
@@ -367,7 +367,7 @@ The outer key is the backend in use, and the value is a backend-specific configu
 
 ### POST /mount/fs/[path]
 
-Adds a new mount point using the JSON contained in the body. This will return a 405 Method Not Allowed if the mount point already exists.
+Adds a new mount point using the JSON contained in the body. The path is the containing directory, and an `X-File-Name` header should contain the name of the mount. This will return a 405 Method Not Allowed if the mount point already exists.
```
Comment (sellout) [inline README.md]: Is 405 the correct response type here? Because to me, 405 means that you can’t call POST on /mount/fs/foo, when really, you can do that, but the header/content is bad. I think 409 (Conflict) is a better response code. In fact, I think I have it return 409 in the #659 changes, so I can change the docs there.

**Comment by sellout** [comment]
@jdegoes Let me rephrase – I think perhaps my use of “subdirs” was confusing. I was using it in the abstract “virtual filesystem” sense, in that any concrete backend (mongo, aerospike, whatever) is a directory in the virtual filesystem.

So, the single request containing
```json
{ "baz": { "mongodb": … },
  "zap": { "mongodb": … } }
```
is the same as the following two requests
```
X-File-Name: baz

{ "mongodb": … }
```
```
X-File-Name: zap

{ "mongodb": … }
```
